Plots/Satellites/NCL_sat_1.py

```diff
@@ -0,0 +1,114 @@
+"""
+NCL_sat_1.py
+===============
+This script illustrates the following concepts:
+   - Drawing line contours over a satellite map
+   - Manually labeling contours
+   - Transforming coordinates
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/sat_1.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/sat_1_lg.png
+"""
+
+###############################################################################
+# Import packages:
+import xarray as xr
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import numpy as np
+
+import geocat.datafiles as gdf
+import geocat.viz.util as gvutil
+
+###############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine and load the data into xarrays
+ds = xr.open_dataset(gdf.get("netcdf_files/slp.1963.nc"), decode_times=False)
+
+# Get lon/lat data from the 24th timestep
+U = ds.slp[24, :, :]
+
+# Translate short values to float values
+U = U.astype('float64')
+
+# Convert Pa to hPa data
+U = U*0.01
+
+# Fix the artifact of not-shown-data around 0 and 360-degree longitudes
+wrap_U = gvutil.xr_add_cyclic_longitudes(U, "lon")
+
+###############################################################################
+# Create plot
+
+# Set figure size
+fig = plt.figure(figsize=(8, 8))
+
+# Set global axes with an orthographic projection
+proj = ccrs.Orthographic(central_longitude=270, central_latitude=45)
+ax = plt.axes(projection=proj)
+ax.set_global()
+
+# Add land, coastlines, and ocean features
+ax.add_feature(cfeature.LAND, facecolor='lightgray')
+ax.add_feature(cfeature.COASTLINE, linewidth=.5)
+ax.add_feature(cfeature.OCEAN, facecolor='lightcyan')
+ax.add_feature(cfeature.BORDERS, linewidth=.5)
+ax.add_feature(cfeature.LAKES, facecolor='lightcyan', edgecolor='k', linewidth=.5)
+
+# Plot contour data
+p = wrap_U.plot.contour(ax=ax,
+                        transform=ccrs.PlateCarree(),
+                        linewidths=0.5,
+                        levels=30,
+                        cmap='black',
+                        add_labels=False)
+
+# Specify array of contour levels to be labeled- These values were found by setting 'manual'
+# argument in ax.clabel call to 'True' and then hovering mouse over desired location of
+# countour label to find coordinate (which can be found in bottom left of figure window)
+
+# low pressure contour levels- these will be plotted as a subscript to an 'L' symbol
+lowClevels = [(51.54, 169.59), (74.78, 4.54), (60.12, -57.0)]
```

Comment:
The manual placement of contour labels is much easier to follow now. Good job!

I did notice that the subscripts for the low pressure labels are different from the NCL plot. I recommend now that you can specify the coordinates for the labels in lat and lon, find the coordinates for the local pressure minimums and use that to plot the labels. The other labels are drawn where they aesthetically make sense and in a way that makes them legible, but the low pressure labels are drawn over a low and need to be more accurate than estimating. This should help with that.

Reply:
One thing I don't understand about the original NCL plot is how they got the L975 pressure marking. From my understanding, they have a list of contours to be plotted that ranges from 948 to 1064 with a step of 4. So if the list of levels looks like [948, 952, 956... 1064], how could 975 be plotted if it's not in the list? 

![sat 1 ncl script](https://user-images.githubusercontent.com/36773444/85431132-f6fea880-b535-11ea-8b7c-c54281213ac3.PNG)

Reply:
I wrote a helper function to try to look through the xarray data, find the low pressure values, and then use the indexes of those to find the corresponding lat/lon values. This was the result of my most recent commit:

![polyg_19_limited](https://user-images.githubusercontent.com/36773444/85441934-7c895500-b544-11ea-9c60-729df4a1fd90.PNG)

If you have any ideas on what I'm doing wrong/how to get the low pressure coordinates without hard coding in the estimates it would be appreciated.

Reply:
These are the lines that draw the Low Pressure labels, and it appears to be a functionality build into NCL.
![image](https://user-images.githubusercontent.com/42781301/85441982-af8c1280-b55d-11ea-87bc-f8ffa245de2b.png)